data-prepper-plugins/peer-forwarder/src/test/java/com/amazon/dataprepper/plugins/prepper/peerforwarder/discovery/AwsCloudMapPeerListProvider_CreateTest.java

```diff
@@ -72,6 +75,18 @@ void createPeerListProvider_with_missing_required_property(final String property
 
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = {
+            PeerForwarderConfig.AWS_CLOUD_MAP_QUERY_PARAMETERS
+    })
+    void createPeerListProvider_with_missing_optional_map_property(final String propertyToRemove) {
+        pluginSetting.getSettings().remove(propertyToRemove);
+
+        final PeerListProvider result = AwsCloudMapPeerListProvider.createPeerListProvider(pluginSetting, pluginMetrics);
```

Comment:
Thanks! This is the bit of code that I was hoping to test.

The only missing part is to verify that a subsequent call to the AWS SDK has an empty map.

Reply:
The `ServiceDiscoveryAsyncClient` is created as a local variable in the `createPeerListProvider` so there's not a good way to mock it in this flow without introducing PowerMock as a new dependency or a decent chunk of refactoring.

I will drop the private scope from the `getOptionalSettingMap` to test the various scenarios while keeping this PR minimal.